obspy/signal/spectral_estimation.py

```diff
@@ -630,6 +638,78 @@ def current_times_used(self):
         self.__check_histogram()
         return [UTCDateTime(ns=ns) for ns in self._current_times_used]
 
+    def _preload_responses(self):
+        result = []
+        if isinstance(self.metadata, (Inventory, str, Parser)):
+            if isinstance(self.metadata, str):
+                inv = read_inventory(self.metadata)
+            elif isinstance(self.metadata, Parser):
+                inv = read_inventory(BytesIO(self.metadata.get_seed()))
+            else:
+                inv = self.metadata
+            for network in inv.networks:
+                for station in network.stations:
+                    for channel in station.channels:
```

Comment:
should probably replace the nested loop by 

for seed_id in inv.get_contents()['channels']:
 ....